### PR TITLE
Start streaming parts before first completed segment in HLS Low Latency

### PR DESCRIPTION
**Pull request by Jordy84**
In #79 I disabled the randomAccess check to start streaming HLS low latency parts before the first segment was completed. This enabled us to display the video feed much faster as our video source GOP size is 20seconds.

The PR was rejected because it was creating segments without a starting IDR frame. 

This PR solves the same problem in a (hopefully) supported manner.

### CODE REVIEW

**Comment by aler9** [comment]
Hello, by looking at the code, you want to make the playlist available as soon as a certain number of parts is available, is that correct?

This approach may work, but did you test it with iOS?

**Comment by Jordy84** [comment]
Hi @aler9, that is correct. When we release the playlist as soon as a certain number of parts are available, we are able to quickly show the stream even with a GOP size of 20seconds.

I tried my best testing this in the iOS simulator on my Mac using a downloaded video published by ffmpeg. The behaviour is the same as with the main branch only faster. But using both the main branch as my branch the simulator doesn't show video but it does request the correct files. 

Maybe you have some tests you normally do in iOS?

**Comment by Jordy84** [comment]
Hi @aler9,

Is there something I can do to help getting this PR tested/completed? Maybe you know of some tests I can do or write?

Thanks!

**Comment by aler9** [comment]
Hello, as previously written, i have to test this on iOS.

**Comment by jordybsecurite** [comment]
Ok thanks! Will wait for your results. Hopefully it all works :)

**Comment by aler9** [comment]
While this approach works with hls.js, it doesn't work on iOS, that requires at least a complete segment before serving the stream.

There's an alternative way to make video appear immediately (without even waiting `DurationRequiredPartCount`, that would become redundant), that i had in mind for some time, it's just a little difficult to implement, and consists in in replacing gaps with video files that contain black frames, encoded with the same codec of the stream. It would allow immediate playback on any device.

Basically you have to store into the server blank frames encoded with AV1, VP9, H264, H265, generate a 1-second long MP4 segment that contains frames correspondent to the current codec, and serve it instead of gaps. In this way, a playlist is available immediately in full.

I don't know if or when this will be implemented, a PR would be welcome.

**Comment by Jordy84** [comment]
That's really unfortunate.

About the proposed solution, do you mean instead of sending a playlist with 7 non-existent gap.mp4 files, we should immediately send actual blank mp4 files with a length of 1 sec? Seems doable! 

Is it an option to add this PR under a feature flag (enable it using the mediamtx.yaml file) while we develop the other proposed solution? 

Our application really needs the faster startup time right now and we would rather keep using the official images instead of creating and building our own forks. Of course we will commit to developing the more compatible solution
